### PR TITLE
RPC Rework: Client

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

To make the `rpc` package clean, simple and reliable.

## What are the changes implemented in this PR?

- Refactor out `SingleResponseCollector`, `ResponseCollector` interface, `Database`, `RPCExecutor`, `ResponseListener`
- `QueryStreamFuture` and `QueryExecuteFuture` change to `QueryFuture.Stream` and `QueryFuture.Execute`
- `RPCResponse` splits up into `RPCResponse.Ok`, `RPCResponse.Error` and `RPCResponse.Completed`
- Refactor out `GraknClient.session` with default session type and `Session.transaction` with default tx type
- Close transactions and sessions atomically
- Make `RPCTransaction.iterate` and `RPCTransaction.execute` behave similarly to each other
- Set default port to 1729

### CODE REVIEW

**Comment by alexjpwalker** [comment]
Do not merge until grakn core is merged and artifact deployed

File: Grakn.java
Comment (alexjpwalker) [reply Grakn.java]: Ok, updated

File: Grakn.java
Comment (haikalpribadi) [reply Grakn.java:104]: nah that would be weird, @lolski . I know you may think it would be more consistent. It's weird in this case.

Comment (alexjpwalker) [reply Grakn.java:104]: Haikal said `queries` is a bit weird and we should stick with `query`.

**Review by lolski — APPROVED** [review]

**Comment by alexjpwalker** [comment]
The failing test will be resolved in the next PR (it's annoying to debug right now because we have to switch back to Java 8)

**Pull request merged**
